# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1377362663 14400
#      Sat Aug 24 12:44:23 2013 -0400
# Node ID f444e4cef9b9079db246349482f517692810f147
# Parent  8cfc28809a07dee62b76a431dddb7bfe014a07ba
avoid exiting Octave on parse error (bug #39862)

* oct-parse.in.yy (input): Use "parse_error" instead of
"simple_list parse_error".

diff --git a/libinterp/parse-tree/oct-parse.in.yy b/libinterp/parse-tree/oct-parse.in.yy
--- a/libinterp/parse-tree/oct-parse.in.yy
+++ b/libinterp/parse-tree/oct-parse.in.yy
@@ -313,17 +313,17 @@ input           : simple_list '\n'
                     YYACCEPT;
                   }
                 | simple_list END_OF_INPUT
                   {
                     lexer.end_of_input = true;
                     parser.stmt_list = $1;
                     YYACCEPT;
                   }
-                | simple_list parse_error
+                | parse_error
                   { ABORT_PARSE; }
                 ;
 
 simple_list     : opt_sep_no_nl
                   { $$ = 0; }
                 | simple_list1 opt_sep_no_nl
                   { $$ = parser.set_stmt_print_flag ($1, $2, false); }
                 ;
